cpp/include/cudf/io/parquet.hpp

```diff
@@ -138,6 +140,18 @@ class parquet_reader_options {
    */
   [[nodiscard]] bool is_enabled_use_arrow_schema() const { return _use_arrow_schema; }
 
+  /**
+   * @brief Returns true/false depending on whether to read matching projected and filter columns
+   * from mismatched Parquet sources.
+   *
+   * @return `true` if mismatched projected and filter columns will be read from mismatched Parquet
+   * sources.
+   */
+  [[nodiscard]] bool is_enabled_allow_mismatched_pq_schemas() const
```

Comment:
We now also support reading (default on) `arrow_schema` in our Parquet reader so I added the `pq` keyword to better disambiguate between them (though we don't really check for mismatched arrow schemas per se but I thought it would be better to know which schema we are talking about here). I don't really have a strong feeling about `pq` here one way or the other.

Reply:
If you think it helps disambiguate, then sure. I am not familiar enough with the format to know.

Reply:
I don't think any other readers implement this feature. Afaik, this request comes from a use-case in `cudf-polars` where we might want to read some matching columns from otherwise mismatching parquet files. Maybe @wence- can shine some light on it's actual application.

Reply:
I could see this being useful in the case of evolving schemas...newer files add or remove a field, but it's too costly to migrate the old data. This would at least allow queries against the common fields.